### PR TITLE
blockly level 15 almost complete

### PR DESCRIPTION
**Pull request by cpirich**
- Added `endGame` block for blockly in playlab
- Changed `setItemSpeed` block to match the speed values in the JavaScript API (for droplet levels)
- Fixed capitalization on `paramRestrictions` for `playSound`
- Removed `whenTouchAllCharacters` and similar auto-handlers. The `whenGetAll*` versions should be used when `autoDisappear` is enabled
- Created level 15 in hoc2015 blockly. The blocks and events are complete, except the whenGet and whenGetAll events are still missing.

### CODE REVIEW

**Comment by cpirich** [comment]
@breville @islemaster

File: apps/i18n/studio/en_us.json [hunk 1]
```diff
@@ -147,6 +147,9 @@
   "dropletBlock_whenTouchProbot_description": "This function executes when the droid touches Probot characters.",
   "dropletBlock_whenUp_description": "This function executes when the up button is pressed.",
   "emotion": "mood",
+  "endGameWin": "end game as a win",
+  "endGameLose": "end game as a loss",
```
Comment (islemaster) [inline apps/i18n/studio/en_us.json:151]: Sentence case?

Comment (islemaster) [inline apps/i18n/studio/en_us.json:151]: Oh, I see this is consistent. NVM.

File: apps/src/studio/blocks.js [hunk 2]
```diff
@@ -572,7 +572,7 @@ exports.install = function(blockly, blockInstallOptions) {
         .appendTitle(new blockly.FieldDropdown(skin.itemChoices), 'CLASS');
 
       var dropdown = new blockly.FieldDropdown(this.VALUES);
-      dropdown.setValue(this.VALUES[2][1]); // default to slow
+      dropdown.setValue(this.VALUES[1][1]); // default to slow
```
Comment (islemaster) [inline apps/src/studio/blocks.js:575]: Should this now say "default to very slow"?

File: apps/src/studio/dropletConfig.js [hunk 3]
```diff
@@ -70,14 +69,6 @@ module.exports.blocks = [
...
 
-  {func: 'whenTouchAllStormtroopers', block: 'function whenTouchAllStormtroopers() {}', expansion: 'function whenTouchAllStormtroopers() {\n  __;\n}', category: '' },
-  {func: 'whenTouchAllRebelPilots', block: 'function whenTouchAllRebelPilots() {}', expansion: 'function whenTouchAllRebelPilots() {\n  __;\n}', category: '' },
-  {func: 'whenTouchAllPufferPigs', block: 'function whenTouchAllPufferPigs() {}', expansion: 'function whenTouchAllPufferPigs() {\n  __;\n}', category: '' },
-  {func: 'whenTouchAllMynocks', block: 'function whenTouchAllMynocks() {}', expansion: 'function whenTouchAllMynocks() {\n  __;\n}', category: '' },
-  {func: 'whenTouchAllMouseDroids', block: 'function whenTouchAllMouseDroids() {}', expansion: 'function whenTouchAllMouseDroids() {\n  __;\n}', category: '' },
-  {func: 'whenTouchAllTauntauns', block: 'function whenTouchAllTauntauns() {}', expansion: 'function whenTouchAllTauntauns() {\n  __;\n}', category: '' },
-  {func: 'whenTouchAllProbots', block: 'function whenTouchAllProbots() {}', expansion: 'function whenTouchAllProbots() {\n  __;\n}', category: '' },
-
```
Comment (islemaster) [inline apps/src/studio/dropletConfig.js:80]: Are these deletions in dropletConfig.js supposed to be part of this change to the blockly level 15?

File: apps/src/studio/levels.js [hunk 4]
```diff
@@ -3112,4 +3104,65 @@ levels.hoc2015_blockly_14 = utils.extend(levels.js_hoc2015_change_setting,  {
...
+//    'whenGetCharacter': { 'category': 'Events' },
+//
+//    'whenGetAllStormtroopers': { 'category': 'Events' },
+//    'whenGetAllRebelPilots': { 'category': 'Events' },
+//    'whenGetAllPufferPigs': { 'category': 'Events' },
+//    'whenGetAllMynocks': { 'category': 'Events' },
+//    'whenGetAllMouseDroids': { 'category': 'Events' },
+//    'whenGetAllTauntauns': { 'category': 'Events' },
+//    'whenGetAllProbots': { 'category': 'Events' },
+//    'whenGetAllCharacters': { 'category': 'Events' }
```
Comment (islemaster) [inline apps/src/studio/levels.js:3159]: Should this commented code just be removed?

**Comment by islemaster** [comment]
:+1: I like the consolidated, two-parameter commands.

File: apps/src/studio/blocks.js (hunk 2 above)
Comment (cpirich) [inline apps/src/studio/blocks.js:575]: It is stated as "slow" in the UI, but VERY_SLOW in absolute terms. Basically, slow/normal/fast for characters/items are defined to be one step slower than slow/normal/fast for bot speed.

File: apps/src/studio/levels.js (hunk 4 above)
Comment (cpirich) [inline apps/src/studio/levels.js:3159]: These are the list of blocks that are in level 15 droplet that still need to be added to level 15 blockly. I should have been more clear. They will be removed when properly added in the next PR.

File: apps/src/studio/dropletConfig.js (hunk 3 above)
Comment (cpirich) [inline apps/src/studio/dropletConfig.js:80]: The deletions are something that needed to happen after name changes and before trying to recreate level 15 in blockly. (this minimizes that number of things that need to be carried over to blockly). So, not strictly related to the title, but needed.

**Pull request merged**